src/XIVLauncher.Core/Program.cs

```diff
@@ -144,16 +145,18 @@ private static void Main(string[] args)
                 default:
                     throw new PlatformNotSupportedException();
             }
-
-            try
-            {
-                var appId = Config.IsFt == true ? STEAM_APP_ID_FT : STEAM_APP_ID;
-                Steam.Initialize(appId);
-            }
-            catch (Exception ex)
+            if (!Config.IsIgnoringSteam ?? true)
```

Comment:
The logic inside the if condition should probably be done near the other config variable checks under line 87

Reply:
I don't \*think\* that moving it would break anything, (naively not checking what `Secrets = GetSecretProvider(storage);` and `Loc.SetupWithFallbacks();` do), but all I did was wrap the pre-existing logic from the original source in the if condition. If it were to be moved, the entire try-catch block would need to move, right? In which case, creating a new method `void LoadSteam()` with the entire try-catch block would probably make more sense from a code readability standpoint (Keeping `LoadConfig(storage)` isolated to its functionality of setting the stored/default config values in memory).

Reply:
Well thats kind of the point I wanted to get at, doing the config handling stored/default values in config. 

basically appending `Config.IsIgnoringSteam ??= true;` behind line 117 and write the if condition as `if (!Config.IsIgnoringSteam)`. This would move the ?? logic to the rest of the config handling function.

In my mind this would make reading the code a bit easier and cleaner. Functionality wise it shouldn't change anything, I didn't check GetSecretProvider and Loc.SetupWithFallbacks either tho.

Reply:
> basically appending Config.IsIgnoringSteam ??= true; behind line 117 and write the if condition as if (!Config.IsIgnoringSteam). This would move the ?? logic to the rest of the config handling function.

It's on line 100. Using ??= doesn't implicitly cast a bool? to a bool, so either casting or using ?? in the if statement is required or you get `error CS0266: Cannot implicitly convert type 'bool?' to 'bool'. An explicit conversion exists (are you missing a cast?)` and I'm like 99% sure using ?? is the safer option (It'll never try to cast NULL to T to F)

Reply:
Oh yeah, never mind then. You are completely right.